lib/plugins/blocks.js

```diff
@@ -268,6 +268,19 @@ function inject (bot, { version, storageBuilder, hideErrors }) {
     }
   })
 
+
+  // Chunk batches are used by the server to throttle the chunks per tick for players based on their connection speed.
+  let chunkBatchStartTime = 0
+  bot._client.on('chunk_batch_start', (packet) => {
+    chunkBatchStartTime = Date.now()
+  })
+
+  bot._client.on('chunk_batch_finished', (packet) => {
+    const chunkPerMill = packet.batchSize / (Date.now() - chunkBatchStartTime) // Get chunks per millisecond
+    bot._client.write('chunk_batch_received', {
+      chunksPerTick: 25 / chunkPerMill
```

Comment:
what is 25 ?

Reply:
A constant the client uses in that calculation.

Reply:
do you know why?

Reply:
/ what is the source of this ?

Reply:
Well, now I feel dumb and have learned to always double check wiki.vg's info since there is a good chance it could be wrong or slightly incorrect. 

Normal client uses this insanity to get its chunksPerTick.  

https://github.com/extremeheat/extracted_minecraft_data/blob/client1.20.2/client/net/minecraft/client/multiplayer/ChunkBatchSizeCalculator.java 

While the input to the onBatchFinished is the batchsize.  No sure where someone got the 25 from and put it on wiki.vg going to probably correct that information later.

Reply:
Updated to used the same calculation as 1.20.2 Vanilla client uses.  (Although converted to millisecond instead of the nanoseconds they use.)